apps/ws/src/socket/ws.gateway.ts

```diff
@@ -1,41 +1,113 @@
-import { OnGatewayConnection, WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
+const nr = require('newrelic');
+
+import { OnGatewayConnection, OnGatewayDisconnect, WebSocketGateway, WebSocketServer } from '@nestjs/websockets';
 import { Server, Socket } from 'socket.io';
 import { JwtService } from '@nestjs/jwt';
-import { ISubscriberJwt } from '@novu/shared';
+import { ISubscriberJwt, ObservabilityBackgroundTransactionEnum } from '@novu/shared';
 import { SubscriberOnlineService } from '../shared/subscriber-online';
 
 @WebSocketGateway()
-export class WSGateway implements OnGatewayConnection {
+export class WSGateway implements OnGatewayConnection, OnGatewayDisconnect {
   constructor(private jwtService: JwtService, private subscriberOnlineService: SubscriberOnlineService) {}
 
   @WebSocketServer()
   server: Server;
 
+  async handleDisconnect(connection: Socket) {
```

Comment:
Refactored this to utilize the nestjs hook for disconnect instead of managing it by ourself.